common/quotas/global/algorithm/requestweighted.go

```diff
@@ -378,13 +405,16 @@ func (a *impl) Update(id Identity, load map[Limit]Requests, elapsed time.Duratio
 				reduce := snap.missedUpdateScalar(age)
 				next = requests{
 					lastUpdate: snap.now,
-					accepted:   weighted(aps, prev.accepted*reduce, snap.weight),
-					rejected:   weighted(rps, prev.rejected*reduce, snap.weight),
+					// TODO: max(1, actual) so this does not lead to <1 rps allowances?  or maybe just 1+actual and then reduce in used-responses?
+					// otherwise currently this may lead to rare callers getting 0.0001 rps,
+					// and never recovering, despite steady and fair usage.
```

Comment:
I need to spend some more time thinking about the ramifications of saying "calculated rps never goes below 1" but I _suspect_ it'll be something we prefer in practice.

otherwise this currently risks something like:
- 1rps    (for 1s, now has weight of 1)
- nothing (10s, now has weight of 0.00001 or something)
- 1rps    (for 1s, now has weight of 0.00003 or something)
- nothing (for 10s, now has weight of 0.000000005)

^ this is fine with a _single_ host receiving all requests, because it gets the full weight, but if it begins alternating between this host and a new one, the new one will have a weight of 1 vs 0.000000005 and the old host will essentially reject all traffic... and basically never recover because the new and old hosts will never decay into the same order of magnitude (until float precision fails).

that's pretty obviously un-good, but right now this is mitigated by an adjustable ~30s-1m garbage collection.  if it causes too many problems for a domain, we can reduce that further while deciding how to fully address it, and/or just put them back on "local" mode with a higher limit to mitigate the imbalance.